pm: device: runtime: Do not suspend devices when runtime is enabled

ceolin opened the pull request:
- Add API to check if a device has runtime pm enabled
- Do not suspend a device if it is using device runtime

Commit:
pm: device: runtime: Add API to check if it is enabled

Add a new API to check if a device has device runtime feature enabled.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
pm: Handle devices with device runtime enabled

Do not suspend a device if it has device runtime enabled.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

gmarull commented on include/pm/device_runtime.h:
```diff
@@ -116,12 +116,27 @@ int pm_device_runtime_put(const struct device *dev);
+ *
+ * @param dev Device instance.
+ *
+ * @retval true If device has enabled device runtime.
```
`If the device has device runtime PM enabled`

gmarull commented on include/pm/device_runtime.h:
```diff
@@ -116,12 +116,27 @@ int pm_device_runtime_put(const struct device *dev);
+ * @param dev Device instance.
+ *
+ * @retval true If device has enabled device runtime.
+ * @retval false If device has not enabled device runtime.
```
`If the device has device runtime PM disabled`

gmarull commented on subsys/pm/power.c:
```diff
@@ -118,7 +119,8 @@ static int pm_suspend_devices(void)
 		int ret;
 
 		/* ignore busy devices */
```
nit: adjust comment here

gmarull reviewed (approved):
LGTM, please adjust docstrings as commented

Commit:
tests: pm: power_mgmt: Check device runtime pm integration

Check if the system power management does not interfere with a device
when it has enabled device runtime PM.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>
